programs/server/Server.cpp

```diff
@@ -773,7 +773,51 @@ try
     LOG_INFO(log, "Available CPU instruction sets: {}", cpu_info);
 #endif
 
-    bool will_have_trace_collector = hasPHDRCache() && config().has("trace_log");
+    bool has_trace_collector = false;
+    /// Disable it if we collect test coverage information, because it will work extremely slow.
+#if !WITH_COVERAGE
+    /// Profilers cannot work reliably with any other libunwind or without PHDR cache.
+    if (hasPHDRCache() && config().has("trace_log"))
+    {
+        has_trace_collector = true;
+
+        /// Set up server-wide memory profiler (for total memory tracker).
+        if (server_settings.total_memory_profiler_step)
```

Comment:
maybe this needs to be moved after `createTraceCollector`, not sure how it works but just in case to avoid sending through unopened pipe